components/coprv2_plugin_api/src/plugin_api.rs

```diff
@@ -0,0 +1,73 @@
+// Copyright 2021 TiKV Project Authors. Licensed under Apache-2.0.
+
+use super::storage_api::*;
+use std::any::Any;
+
+/// Raw bytes of the request payload from the client to the coprocessor.
+pub type RawRequest = [u8];
+/// The response from the coprocessor encoded as raw bytes that are sent back to the client.
+pub type RawResponse = Vec<u8>;
+
+/// A plugin that allows users to execute arbitrary code on TiKV nodes.
+///
+/// If you want to implement a custom coprocessor plugin for TiKV, your plugin needs to implement
+/// the [`CoprocessorPlugin`] trait.
+pub trait CoprocessorPlugin: Any + Send + Sync {
+    /// Returns the name of the plugin.
+    /// Requests that are sent to TiKV coprocessor must have a matching `copr_name` field.
+    fn name(&self) -> &'static str;
+
+    /// A callback fired immediately after the plugin is loaded. Usually used
+    /// for initialization.
+    ///
+    /// The default implementation does nothing, but can be overridden by plugins.
+    fn on_plugin_load(&self) {}
```

Comment:
If the `$constructor` in `declare_plugin!` has done the initialization, what is the difference between them?

Reply:
hm good point, I had in mind that `$constructor` really just creates the object and `on_plugin_load()` does some (optional) initialization - so just to make a distinction here and maybe to make a clearer API (same for `on_plugin_unload()` and `Drop`). But you are right: it might be confusing because they are the same. I'll remove the functions and instead add a comment :)